src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/MethodToStateMachineRewriter.cs

```diff
@@ -621,6 +622,52 @@ public override BoundNode VisitBlock(BoundBlock node)
             return PossibleIteratorScope(node.Locals, () => (BoundStatement)base.VisitBlock(node));
         }
 
+        public override BoundNode VisitScope(BoundScope node)
```

Comment:
I assume each variable gets assigned a scope, then that scope will get added to the various dictionaries, where it will then be turned into a lambda frame.

So my reading here is that your change "just works" in the sense that your new BoundScope is chosen as the appropriate scope node and the lambda frame is built off that (including the introduction of the various proxies).